derive/src/rpc_trait.rs

```diff
@@ -156,12 +158,15 @@ fn generate_rpc_item_trait(item_trait: &syn::ItemTrait) -> Result<(syn::ItemTrai
 		rpc_trait.has_metadata,
 		has_pubsub_methods,
 	)?;
+
+	let rpc_client_module = generate_client_module(&method_registrations, &item_trait)?;
```

Comment:
Could we pass in `DeriveOptions` and only generate the client/server modules where necessary? If the impact on compile time is negligible then it should be fine. But we have some pretty big rpc traits e.g. in `parity-ethereum`.